packages/react-i18n/src/index.tsx

```diff
@@ -12,18 +14,80 @@ export interface I18nReact {
 	_x: I18n[ '_x' ];
 	isRTL: I18n[ 'isRTL' ];
 	i18nLocale: string;
+	localeData?: LocaleData;
+	hasTranslation: ( singular: string, context?: string ) => boolean;
+	addFilter: typeof addFilter;
+	removeFilter: typeof removeFilter;
+	applyFilters: typeof applyFilters;
+}
+
+/**
+ * Private hooks instance.
+ */
+const hooks = createHooks();
```

Comment:
This creates a singleton that's shared by all `i18n` context providers inside the same app. Then the contexts can't be completely independent from each other. That's a problem in server-side SSR usage, where multiple React pages can be rendered for mutliple users with different locales at the same time.

Reply:
Thank you for explaining, it seems like I misunderstood the problem with SSR initially, but it's all clear now. Fixed it in https://github.com/Automattic/wp-calypso/pull/44743/commits/083ec10ff6a9b5ceb4d776c941b6447aedbf7c62